drivers/eeprom/eeprom_mb85rs1mt.c

```diff
@@ -0,0 +1,303 @@
+}
+
+static int eeprom_mb85rs1mt_wrdi(const struct device *dev)
+{
+	const struct eeprom_mb85rs1mt_config *config = dev->config;
+	uint8_t cmd = EEPROM_MB85RS1MT_WRDI;
+	const struct spi_buf tx_buf = {
+		.buf = &cmd,
+		.len = sizeof(cmd),
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1,
+	};
+
+	return spi_write_dt(&config->spi, &tx);
+}
+
+static int eeprom_mb85rs1mt_write(const struct device *dev, off_t offset, const void *buf,
+				  size_t len)
+{
+	const struct eeprom_mb85rs1mt_config *config = dev->config;
+	struct eeprom_mb85rs1mt_data *data = dev->data;
+	uint8_t cmd[4] = {EEPROM_MB85RS1MT_WRITE, 0, 0, 0};
+	uint8_t *paddr = &cmd[1];
+	int err;
+
+	if (config->readonly) {
+		LOG_ERR("attempt to write to read-only device");
+		return -EACCES;
+	}
+
+	if (offset + len > config->size) {
+		LOG_ERR("attempt to write past device boundary");
+		return -EINVAL;
+	}
+
+	/* Populate address in command */
+	*paddr++ = (offset >> 16) & 0xFF;
+	*paddr++ = (offset >> 8) & 0xFF;
+	*paddr++ = offset & 0xFF;
+
+	const struct spi_buf tx_bufs[2] = {
+		{
+			.buf = cmd,
+			.len = sizeof(cmd),
+		},
+		{
+			.buf = (void *)buf,
+			.len = len,
+		},
+	};
+	const struct spi_buf_set tx = {
+		.buffers = tx_bufs,
+		.count = ARRAY_SIZE(tx_bufs),
+	};
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	err = eeprom_mb85rs1mt_wren(dev);
+	if (err < 0) {
+		LOG_ERR("failed to disable write protection (err %d)", err);
+		k_mutex_unlock(&data->lock);
+		return err;
+	}
+
+	err = spi_write_dt(&config->spi, &tx);
+	if (err < 0) {
+		LOG_ERR("failed to write to FRAM (err %d)", err);
+		k_mutex_unlock(&data->lock);
+		return err;
+	}
+
+	err = eeprom_mb85rs1mt_wrdi(dev);
+	if (err < 0) {
+		LOG_ERR("failed to disable write (err %d)", err);
+	}
+
+	k_mutex_unlock(&data->lock);
+
+	return err;
+}
+
+static size_t eeprom_mb85rs1mt_size(const struct device *dev)
+{
+	const struct eeprom_mb85rs1mt_config *config = dev->config;
+
+	return config->size;
+}
+
+static int eeprom_mb85rs1mt_rdid(const struct device *dev)
+{
+	const struct eeprom_mb85rs1mt_config *config = dev->config;
+	struct eeprom_mb85rs1mt_data *data = dev->data;
+	uint8_t id[4];
+	uint8_t cmd = EEPROM_MB85RS1MT_RDID;
+	int err;
+
+	const struct spi_buf tx_buf = {
+		.buf = &cmd,
+		.len = sizeof(cmd),
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1,
+	};
+	const struct spi_buf rx_bufs[2] = {
+		{
+			.buf = NULL,
+			.len = sizeof(cmd),
+		},
+		{
+			.buf = id,
+			.len = sizeof(id),
+		},
+	};
+	const struct spi_buf_set rx = {
+		.buffers = rx_bufs,
+		.count = ARRAY_SIZE(rx_bufs),
+	};
+	k_mutex_lock(&data->lock, K_FOREVER);
+	err = spi_transceive_dt(&config->spi, &tx, &rx);
+	k_mutex_unlock(&data->lock);
+
+	if (err < 0) {
+		LOG_ERR("failed to read RDID (err %d)", err);
+		return err;
+	}
+
+	/* Validate Manufacturer ID and Product ID */
+	if (id[0] != 0x04 || id[1] != 0x7F || id[2] != 0x27 || id[3] != 0x03) {
```

Comment:
Lower 5 bits of id[2] serve as density, you probably do not want to strictly expect 0x27, 1 megabit mem, as the same driver should, mostly, properly support mb85rs2, which is 2 megabit device and has 8 instead of 7 in the identifier.

Reply:
Should be fixed now.

I've also renamed the whole thing to `mb85rsxx` so that we don't specify the storage size right in the driver's name.